flake.nix

```
@@ -0,0 +1,42 @@
+{
+  description = "Rust devshell for Iced";
+
+  inputs = {
+    nixpkgs.url      = "github:NixOS/nixpkgs/nixos-unstable";
+    rust-overlay.url = "github:oxalica/rust-overlay";
+    flake-utils.url  = "github:numtide/flake-utils";
+  };
+
+  outputs = { nixpkgs, rust-overlay, flake-utils, ... }:
+    flake-utils.lib.eachDefaultSystem (system:
+      let
+        overlays = [ (import rust-overlay) ];
+        pkgs = import nixpkgs {
+          inherit system overlays;
+        };
+      in
+      with pkgs;
+      {
+        devShells.default = mkShell rec {
+          nativeBuildInputs = [
+            bacon
+            cargo-udeps
+            cargo-edit
+            rust-analyzer
+            cargo-deny
+            rust-bin.stable.latest.default
+
+            pkg-config
+            dbus
+          ];
+          buildInputs = [
+            libxkbcommon
+          ];
+          runtimeDependencies = [
+            wayland
+          ];
```

These three sections can simple be `packages` or `buildInputs`. 

Since it's not only for Linux users, I assume macOS (as myself) users will also tune in, we have to put linux only dependencies behind a condition.

```suggestion
          buildInputs = [
            bacon
            cargo-udeps
            cargo-edit
            rust-analyzer
            cargo-deny
            rust-bin.stable.latest.default
            pkg-config
            dbus
            libxkbcommon
          ] ++ lib.optionals (stdenv.isLinux) [
            wayland
          ];
```